### ISSUE TITLE
Auto-grouping breaks when using Metropolis theme

### ISSUE THREAD

**Issue by Krzmbrzl** [pdfpc/pdfpc#620]
**Expected Behaviour**
Auto-grouping to work or at least not reorder slides completely

**Actual Behaviour**
When using the Metropolis theme, for each new section in the presenetation a auto-generated cover-slide can be inserted. These seem to confuse the autogrouping algorithm since these coverslides appear out-of-order in pdfpc's preview and are then missing in said preview once they would actually be shown.

This is confusing since first you expect the next slide to be a cover-slide when in fact there are yet more slides of the first section (and upon switching to the next slide those are indeed shown instead of the previewed cover-slide) and then later you see a preview of the first slide of section 2 but upon switching you are now displaying the cover-slide instead.

**Steps to reproduce**

Compile this example presentation:
```latex
\documentclass[aspectratio=169]{beamer}

\usetheme[sectionpage=progressbar, numbering=fraction, progressbar=frametitle]{metropolis}
\setbeamertemplate{section in toc}[sections numbered]


\author{The author}
\date{\today}
\title{An example}

\begin{document}
	\maketitle

	{
		\metroset{numbering=none}

		\begin{frame}[noframenumbering]{Outline}
			\tableofcontents
		\end{frame}
	}

	{
		\metroset{sectionpage=none}
		\section{Introduction}

		\subsection{Overview: Automation frameworks}
	}

	\begin{frame}
		test

		\pause

		test2

		\pause

		test3
	\end{frame}

	\section{Second}
	\begin{frame}{Second}
		test

		\pause

		test2

		\pause

		test3
	\end{frame}

\end{document}
```
and then run pdfpc on this to see for yourself.

pdfpc version: 4.3.4
Used Distribution (GTK version, vala version): Ubuntu 20.04
Display Server (X/Wayland): X
HiDPI Setting (yes/no): no

----

Maybe if you could briefly outline the algorithm that is used for auto-grouping slides, I might be bale to assist in finding a proper solution to this issue.

**Comment by fnevgeny** [comment]
You can use, e.g., https://github.com/HeimMatthias/pdfpagelabels to see the labels `beamer` produces for this document:
` 0 0 1 1 1 1 2 2 2`. That is, the first "user slide" contains 2 overlays, 2nd - 4, and the last one has 3. And that's how `pdfpc` shows them. BTW, please upgrade to the latest released version (4.5), which fixes a zero-indexed-page bug #355.

**Comment by fnevgeny** [comment]
Of course, feel free to reopen this ticket if you find out that the current version still doesn't work as you expect.

**Comment by jolars** [comment]
Sorry for dredging up this old issue, but I have been affected by this for a while and didn't know what was going on. This is not directly caused by anything in metropolis (or moloch, which is a fork that I am maintaining). Or rather, it's related to the `noframenumbering` setting, which I guess was the point discovered in #623. Here's a minimal working example:

```latex
\documentclass{beamer}

\title{Title}

\AtBeginSection[]{
  \begin{frame}[noframenumbering]
    \insertsectionhead
  \end{frame}
}

\begin{document}

\maketitle

\begin{frame}
  \frametitle{Frame One}

  Some text \pause

  Some other text \pause

  More text

\end{frame}

\begin{frame}
  \frametitle{Frame Two}

  Frame content

\end{frame}

\section{Section Title}

\begin{frame}
  \frametitle{Another Frame Title}

  More frame content

\end{frame}

\end{document}
```

With the defaults, I see:

<img width="973" height="552" alt="Image" src="https://github.com/user-attachments/assets/12273f2b-d1d1-4d76-ba30-671608a88acc" />

The preview shows "Section Title", but actually "Frame Two" is up next.

After  #623, you can fix this by:

```
option next-slide-first-overlay true
```

in the `pdfpcrc` config file. 

I thought I would just put this here if anyone stumbled on this issue again.

I guess it would be nice if this was just working as expected without having to rely on this option, but I am guessing that that's hard to solve.
